Libraries/Utilities/EnumerableExtensions.cs

```diff
@@ -7,20 +7,23 @@ public static class EnumerableExtensions
         /// <summary>
         /// Awaits the tasks sequentially. An alternative to
         /// <see cref="Task.WhenAll(System.Collections.Generic.IEnumerable{System.Threading.Tasks.Task})"/> and
-        /// <code>Nito.AsyncEx.TaskExtensions.WhenAll</code> when true multi-threaded asynchronicity is not desirable.
+        /// <c>Nito.AsyncEx.TaskExtensions.WhenAll</c> when true multi-threaded asynchronicity is not desirable.
         /// </summary>
         /// <param name="source">A collection of items.</param>
-        /// <param name="asyncOperation">An <code>async</code> function to call on each item.</param>
+        /// <param name="asyncOperation">An <see langword="async"/> function to call on each item.</param>
         /// <typeparam name="TItem">The type of the input collection's items.</typeparam>
         /// <typeparam name="TResult">The type of the output collection's items.</typeparam>
         /// <returns>When awaited the task contains the results which were added one-by-one.</returns>
-        public static async Task<IList<TResult>> AwaitEach<TItem, TResult>(
+        public static async Task<IList<TResult>> AwaitEachAsync<TItem, TResult>(
             this IEnumerable<TItem> source,
             Func<TItem, Task<TResult>> asyncOperation)
         {
             var results = new List<TResult>();
             foreach (var item in source) results.Add(await asyncOperation(item));
             return results;
         }
+
+        public static IList<T> AsList<T>(this IEnumerable<T> collection) =>
+            collection is IList<T> list ? list : new List<T>(collection);

```

Comment:
Nem ugyanezt csinálja a LINQ `ToList()` is?

Reply:
A ToList sose castol, mindig új listát csinál

Reply:
Aha. Akkor viszont ezt commenteld ide légyszi.

Reply:
ja, ez is megvolt xmldocban

Reply:
Ja, láttam.